### DISCUSSION TITLE
BIRT runtime install on RHEL for shell generation

### DISCUSSION THREAD

**Discussion by slhotellier** [eclipse-birt/birt#1221] [Q&A] [answered]
Hello

I am trying to install BIRT runtime engine on a RHEL8.
Unfortunately the documentation is not clear enough to me.
I updated the genReport.sh file to put my environment variables there:
export WORK_DIR=/opt/birt
export BIRTCLASSPATH=$BIRT_HOME/ReportEngine/lib
export CLASSPATH=$BIRT_HOME/ReportEngine/lib

But I got an error: Could not find or load main class org.eclipse.birt.report.engine.api.ReportRunner

If someone would be kind enough to help me, I would be happy to enrich the documentation.

Best regards

**Comment by hvbtup** [comment]
What is your environment? Are you using Java 8 perhaps? BIRT needs Java >= 11.

**Reply by slhotellier** [reply]
thanks for your response

I was in java 8 so I moved to java 11 but the problem is still there : 

#/opt/birt/birt-runtime/ReportEngine/genReport.sh /opt/birt/workspace/rapport/rapport\ de\ nuit.rptdesign -o toto.pdf -f PDF

BIRT_HOME=/opt/birt/birt-runtime
WORK_DIR=/opt/birt
BIRTCLASSPATH=/opt/birt/birt-runtime/ReportEngine/lib
JAVACMD=/usr/lib/jvm/java-11-openjdk-11.0.18.0.10-2.el8_7.x86_64/bin/java
Error: Could not find or load main class org.eclipse.birt.report.engine.api.ReportRunner
Caused by : java.lang.ClassNotFoundException: org.eclipse.birt.report.engine.api.ReportRunner

**Comment by hvbtup** [comment]
The `genReports.sh` script seems to be broken/incomplete in comparison to its windows counterpart `genReport.bat`.
Not many people know that, but Java supports a class-path entry ending in a star `*`, which means all *.zip and *.jar files in the given directory.
Well, on Linux probably the shell would already expand the star with more or less the same effect (unless the directory contains superfluous garbage files).

You should compare the environment variables to those of the windows version and adjust them.
In particular, append `/*`to `ReportEngine/lib`.

**Comment by slhotellier** [comment]
thanks a lot 
it's better with * at the end of the lib directory.

I can now generate the sample report.

**Reply by hvbtup** [reply]
Do you think you can create a PR with a patched script?

**Comment by slhotellier** [comment]
Yes off course, I will finish my test today and create a PR with a documented patch script

**Comment by slhotellier** [comment, marked as answer]
Sorry I can't find the genReport.sh in the project so I can't make the RP 

this is the new genReport.sh I would send : 


```
#!/bin/sh

################ USAGE OF REPORTRUNNER #########################
# org.eclipse.birt.report.engine.impl.ReportRunner Usage:
# --mode/-m [ run | render | runrender] the default is runrender
# for runrender mode:
# we should add it in the end <design file>
#    --format/-f [ HTML | PDF ]
#    --output/-o <target file>
#    --htmlType/-t < HTML | ReportletNoCSS >
#    --locale/-l<locale>
#    --parameter/-p <"parameterName=parameterValue">
#    --file/-F <parameter file>
#    --encoding/-e <target encoding>
#
# Locale: default is English
# parameters in command line will overide parameters in parameter file
# parameter name cannot include characters such as \' \'\, \'=\'\, \':\'
#
# For RUN mode:
# we should add it in the end <design file>
#    --output/-o <target file>
#    --locale/-l<locale>
#    --parameter/-p <parameterName=parameterValue>
#    --file/-F <parameter file>
#
# Locale: default is English
# parameters in command line will overide parameters in parameter file
# parameter name cannot include characters such as \' \'\, \'=\'\, \':\'
#
# For RENDER mode:
# we should add it in the end <design file>
#    --output/-o <target file>
#    --page/-p <pageNumber>
#    --locale/-l<locale>
#
# Locale: default is English
################ END OF USAGE #########################

# echo set common variables
#export BIRT_HOME=BIRT-runtime_install_dir
export BIRT_HOME=/opt/birt/birt-runtime
export WORK_DIR=$PWD
# class path : don't forget to and your connection drivers in the classpath directory
export BIRTCLASSPATH=$BIRT_HOME/ReportEngine/lib/*

echo BIRT_HOME=$BIRT_HOME
echo WORK_DIR=$WORK_DIR
echo BIRTCLASSPATH=$BIRTCLASSPATH

if [ "$BIRT_HOME" = "" ]

then
echo "BIRT_HOME must be set before ReportRunner can run";
else

java_io_tmpdir=$WORK_DIR/tmpdir
org_eclipse_datatools_workspacepath=$java_io_tmpdir/workspace_dtp
mkdir -p $org_eclipse_datatools_workspacepath

# JAVA version >= 11
JAVACMD='java';
$JAVACMD -Djava.awt.headless=true -cp "$BIRTCLASSPATH" -DBIRT_HOME="$BIRT_HOME" -Dorg.eclipse.datatools_workspacepath="$org_eclipse_datatools_workspacepath" org.eclipse.birt.report.engine.api.ReportRunner ${1+"$@"}

fi
```

**Reply by hvbtup** [reply]
please send this as a file attachment (with a suffix .txt)

**Reply by syc795** [reply]
https://github.com/eclipse/birt/edit/master/build/org.eclipse.birt.releng.birtbuilder/extras/genReport.sh

**Comment by slhotellier** [comment]
[genReport.sh.txt](https://github.com/eclipse/birt/files/10849434/genReport.sh.txt)

**Comment by slhotellier** [comment]
pull request #1222
